runtime/service.go

```diff
@@ -681,6 +668,33 @@ func (s *service) buildVMConfiguration(req *proto.CreateVMRequest) (*firecracker
 	return &cfg, nil
 }
 
+func (s *service) buildNonStubDrives(req *proto.CreateVMRequest) []models.Drive {
+	var builder firecracker.DrivesBuilder
+
+	if input := req.RootDrive; input != nil {
+		builder = firecracker.DrivesBuilder{}.
+			WithRootDrive(input.PathOnHost, func(d *models.Drive) {
+				d.IsReadOnly = firecracker.Bool(input.IsReadOnly)
```

Comment:
If `input.IsReadOnly` is unset in the request, does this evaluate to true or false?

Reply:
Currently `false` which is inconsistent with the configuration file parsing.

While proto2 is having a way to [specify per-field default values](https://developers.google.com/protocol-buffers/docs/proto#optional), proto3 [doesn't have that anymore](https://developers.google.com/protocol-buffers/docs/proto3#default).

> Note that for scalar message fields, once a message is parsed there's no way of telling whether a field was explicitly set to the default value (for example whether a boolean was set to false) or just not set at all: you should bear this in mind when defining your message types. For example, don't have a boolean that switches on some behaviour when set to false if you don't want that behaviour to also happen by default.

Reply:
Perhaps we can change the name of the field, so it follows the inverse, like `IsWritable`? That way it defaults to false. I don't know if I like that, but I think it beats using a string or enumerables.

Reply:
Sounds good. Will do.

Reply:
Fixed on 91938abf83568d853544632e213b73c12d1a1b70

However, can't we just make IsReadOnly required?

Reply:
Another option is making the root drive always read-only, since https://github.com/firecracker-microvm/firecracker-containerd/blob/master/docs/drive-mounts-proposal.md will separate root drives from non-root drives.

Reply:
@kzys - You can't force the requirement though. How do you know if the user meant to set `IsReadOnly` to false versus not setting it at all?

> Another option is making the root drive always read-only, since https://github.com/firecracker-microvm/firecracker-containerd/blob/master/docs/drive-mounts-proposal.md will separate root drives from non-root drives.

I'd be more in favor of this if we want to get rid of the field. However, if someone ever wanted a writable root drive, then the `IsWritable` would be the implementation preferred. So you either can not do the work now, but we will probably need to do it in the future.

Reply:
I meant, `IsReadOnly` required on Protobuf, instead of making the field optional. The change I'm making is small, but the way we workaround the mismatch between JSON vs. Protobuf seems hacky.

Reply:
proto3 does not have the concept of optional versus required. proto2 does, but we don't use that.

Reply:
Wow. Sorry. I haven't noticed that proto3 removed that aw well.

Reply:
6690f80e0383c6061290e93ac461489860565a20 renames IsReadOnly to IsWritable